api/src/routes/identity/index.ts

```diff
@@ -19,12 +19,19 @@ export class IdentityRoutes {
 		private readonly logger: ILogger
 	) {}
 
-	createIdentity = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+	createIdentity = async (req: Request, res: Response, next: NextFunction) => {
 		try {
 			const createIdentityBody: CreateIdentityBody = req.body;
+
```

Comment:
I think we should write and capture exception instead of doing read/write.
Read and insert can lead to race conditions, while the simple createIdentity is atomic.

Reply:
I think it is better to check otherwise we always create a request to the tangle creating an identity which we then cannot add since we receive an exception by the database later. So checking it in advance would avoid unnecessary tangle requests

Reply:
I agree